hub/api/dataset_handler.py

```diff
@@ -0,0 +1,180 @@
+from hub.util.exceptions import DatasetHandlerError, PathNotEmptyException
+from hub.util.get_storage_provider import get_storage_provider
+from typing import Optional
+from hub.constants import DEFAULT_LOCAL_CACHE_SIZE, DEFAULT_MEMORY_CACHE_SIZE
+from .dataset import Dataset
+from hub.util.keys import dataset_exists
+
+
+class dataset:
+    def __new__(
+        cls,
+        path: str,
+        read_only: bool = False,
+        overwrite: bool = False,
+        public: Optional[bool] = True,
+        memory_cache_size: int = DEFAULT_MEMORY_CACHE_SIZE,
+        local_cache_size: int = DEFAULT_LOCAL_CACHE_SIZE,
+        creds: Optional[dict] = None,
+    ):
+        """Returns a Dataset object referencing either a new or existing dataset.
+
```

Comment:
```suggestion
        """Returns a Dataset object referencing either a new or existing dataset.

    Important:
        Using `overwrite` will delete all of your data if it exists! Be very careful when setting this parameter.

```

Reply:
i think we should add a check here that if a dataset exceeds a certain size (like say 1GB), we should raise an error and require them to use `hub.delete` or any explicit deletion method.

Reply:
(btw im not saying to implement this now)

Reply:
pls commit the suggested docstring